sympy/functions/elementary/tests/test_complexes.py

```diff
@@ -870,7 +870,7 @@ def test_derivatives_issue_4757():
     assert im(f(x)).diff(x) == im(f(x).diff(x))
     assert re(f(y)).diff(y) == -I*im(f(y).diff(y))
     assert im(f(y)).diff(y) == -I*re(f(y).diff(y))
-    assert Abs(f(x)).diff(x).subs(f(x), 1 + I*x).doit() == x/sqrt(1 + x**2)
+    assert Abs(f(x)).diff(x).subs(f(x), 1 + I*x).doit().simplify() == x/sqrt(1 + x**2)
```

Comment:
I'm not sure exactly what causes the change here but I find it surprising that on master this just happens to simplify automatically:
```python
In [4]:     x = Symbol('x', real=True)
   ...:     y = Symbol('y', imaginary=True)
   ...:     f = Function('f')

In [5]: Abs(f(x)).diff(x)
Out[5]: 
⎛           ⎛d       ⎞              ⎛d       ⎞⎞           
⎜re(f(x))⋅re⎜──(f(x))⎟ + im(f(x))⋅im⎜──(f(x))⎟⎟⋅sign(f(x))
⎝           ⎝dx      ⎠              ⎝dx      ⎠⎠           
──────────────────────────────────────────────────────────
                           f(x)                           

In [6]: Abs(f(x)).diff(x).subs(f(x), 1+I*x)
Out[6]: 
⎛    ⎛d          ⎞     ⎛d          ⎞⎞              
⎜x⋅im⎜──(ⅈ⋅x + 1)⎟ + re⎜──(ⅈ⋅x + 1)⎟⎟⋅sign(ⅈ⋅x + 1)
⎝    ⎝dx         ⎠     ⎝dx         ⎠⎠              
───────────────────────────────────────────────────
                      ⅈ⋅x + 1                      

In [7]: Abs(f(x)).diff(x).subs(f(x), 1+I*x).doit()
Out[7]: 
     x     
───────────
   ________
  ╱  2     
╲╱  x  + 1 
```
This actually happens because:
```python
In [8]: sign(I*x+ 1).doit()
Out[8]: 
  ⅈ⋅x + 1  
───────────
   ________
  ╱  2     
╲╱  x  + 1 
```
That's different in this PR:
```python
In [2]: sign(I*x+ 1).doit()
Out[2]: sign(ⅈ⋅x + 1)
```
That's due to this giving None in the PR:
```python
In [2]: (I*x+ 1).is_zero
Out[2]: False
```

Reply:
And this still works with the mods:
```python
>>> x = Symbol('x', real=True)
>>> y = Symbol('y', imaginary=True)
>>> Abs(f(x)).diff(x).subs(f(x), 1 + I*x).doit()
x/sqrt(x**2 + 1)
```